test/CSModule.t.sol

```diff
@@ -7224,6 +7224,297 @@ contract CSMDepositableValidatorsCount is CSMCommon {
+        uint256 depositableBefore = csm
+            .getNodeOperator(noId)
+            .depositableValidatorsCount;
+
+        assertEq(depositableBefore, 7);
+
+        updateToBetterCurve();
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore
+        );
+
+        csm.normalizeQueue(noId);
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore
+        );
+    }
+
+    function test_depositableOnly_UpdateToWorseCurve() public {
+        uint256 noId = createNodeOperator(7);
+
+        uint256 depositableBefore = csm
+            .getNodeOperator(noId)
+            .depositableValidatorsCount;
+
+        assertEq(depositableBefore, 7);
+
+        updateToWorseCurve();
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore
+        );
+
+        csm.normalizeQueue(noId);
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore - 2
+        );
+    }
+
+    function test_partiallyUnbondedDepositedOnly_UpdateToBetterCurve() public {
+        uint256 noId = createNodeOperator(7);
+        csm.obtainDepositData(7, "");
+        uint256 depositableBefore = csm
+            .getNodeOperator(noId)
+            .depositableValidatorsCount;
+        assertEq(depositableBefore, 0);
+
+        csm.reportELRewardsStealingPenalty(
+            noId,
+            blockhash(block.number),
+            1 ether
+        );
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore
+        );
+
+        updateToBetterCurve();
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore
+        );
+
+        csm.normalizeQueue(noId);
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore
+        );
+    }
+
+    function test_partiallyUnbondedDepositedOnly_UpdateToWorseCurve() public {
+        uint256 noId = createNodeOperator(7);
+        csm.obtainDepositData(7, "");
+        uint256 depositableBefore = csm
+            .getNodeOperator(noId)
+            .depositableValidatorsCount;
+        assertEq(depositableBefore, 0);
+
+        csm.reportELRewardsStealingPenalty(
+            noId,
+            blockhash(block.number),
+            1 ether
+        );
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore
+        );
+
+        updateToWorseCurve();
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore
+        );
+
+        csm.normalizeQueue(noId);
+        assertEq(
+            csm.getNodeOperator(noId).depositableValidatorsCount,
+            depositableBefore
+        );
```

Comment:
Add unbonded assertions. Not sure if normalizing queue when all are bonded makes sense